### PR TITLE
feat: comment feed

### PR DESCRIPTION
**Pull request by rebelchris**
### Describe what this PR does
- Added additional comment feed for users enrolled in this experiment
- It's the quick-n-dirty implementation but tried to keep the mess to a minimum. (Ideally lean feed would make this a bit easier to organize, took some learnings from here)
- Sidebar menu has a `new` badge until you click it (uses actions)
- Refactored profile replies screen to re-use this feed

Screenshot 2024-03-22 at 13 18 25

Placeholder vs loaded
Screenshot 2024-03-22 at 13 00 21

Try it out at: (Please wait till API is merged https://github.com/dailydotdev/daily-api/pull/1767)
https://preview3.app.daily.dev/

## Events

Did you introduce any new tracking events?
Don't forget to update the [Analytics Taxonomy sheet](https://docs.google.com/spreadsheets/d/18Lv7zXges9QfVX5VYL1a-Hyl0e1sQ3sLr0OK8YZWKXI/edit#gid=0)

Log the new/changed events below:

| Type   | event_name  | value |
|--------|-------------|-------|
| Change/New | event name  | extra: { ... } |

### On those affected packages:
- [ ] Have you done sanity checks in the webapp?
- [ ] Have you done sanity checks in the extension?
- [ ] Does this not break anything in companion?

### Did you test the modified components media queries?
- [ ] MobileL (420px)
- [ ] Tablet (656px)
- [ ] Laptop (1020px)

#### Did you test on actual mobile devices?
- [ ] iOS (Chrome and Safari)
- [ ] Android

AS-202 #done

### CODE REVIEW

File: packages/shared/src/components/MainFeedLayout.tsx [hunk 1]
```diff
@@ -240,24 +256,43 @@ export default function MainFeedLayout({
...
         />
+      ) : (
+        feedProps && (
+          <Feed
+            {...feedProps}
+            className={classNames(
+              shouldUseMobileFeedLayout && !isFinder && 'laptop:px-6',
+            )}
+          />
+        )
```
Comment (DragosIorgulescu) [inline packages/shared/src/components/MainFeedLayout.tsx:295]: More for my understanding, I am not sure I understand why this change is needed if the comment feed is just added to the user replies page?

File: packages/shared/src/components/sidebar/DiscoverSection.tsx [hunk 2]
```diff
@@ -51,6 +50,40 @@ export function DiscoverSection({
...
+  } else {
+    discoverMenuItems.splice(2, 0, {
+      icon: (active: boolean) => (
+        <ListIcon Icon={() => <DiscussIcon secondary={active} />} />
+      ),
+      title: 'Best discussions',
+      path: '/discussed',
+      action: () => onNavTabClick?.('discussed'),
+    });
+  }
```
Comment (DragosIorgulescu) [inline packages/shared/src/components/sidebar/DiscoverSection.tsx:85]: lacking a bit of context here, seems we are marking the `/discussed` page as new? I don't see any modifications to the page though, or is this just to register the action?

File: packages/shared/src/hooks/useFeedLayout.ts [hunk 3]
```diff
@@ -26,18 +29,24 @@ export const useFeedLayout = ({
...
-  const shouldUseMobileFeedLayout = !feedRelated
-    ? isMobile
-    : isMobile && isIncludedFeed;
+
+  const shouldUseCommentFeedLayout =
+    hasCommentFeed && feedName === SharedFeedPage.Discussed;
+  const isMobileAndIncludedFeed = isMobile && isIncludedFeed;
+  const shouldUseMobileFeedLayout = feedRelated
+    ? isMobileAndIncludedFeed
+    : isMobile;
```
Comment (DragosIorgulescu) [inline packages/shared/src/hooks/useFeedLayout.ts:46]: ah, I see now why the `MainFeedLayout` was modified, you're switching to the comments feed on the `/discussed` page if the feature flag is on 👌

**Review by DragosIorgulescu — APPROVED** [review]
Nice, not seeing any issues here 👌

File: packages/shared/src/hooks/useFeedLayout.ts [hunk 4]
```diff
@@ -26,18 +29,24 @@ export const useFeedLayout = ({
   feedName: feedNameProp,
   feedRelated = true,
 }: UseFeedLayoutProps = {}): UseFeedLayout => {
+  const hasCommentFeed = useFeature(feature.commentFeed);
```
Comment (capJavert) [inline packages/shared/src/hooks/useFeedLayout.ts:32]: I think this `useFeature` should be replace with `withExperiment` on `CommentFeed` and then only allocate if users is on `discussions` page.

Comment (capJavert) [reply packages/shared/src/hooks/useFeedLayout.ts:32]: The hook should only return true if user is on discussions. This will prevent user being allocated on any other place where `useFeedLayout` is used.

**Review by capJavert — COMMENTED** [review]
Good work 👌 , approved API so we can test on preview.

File: packages/shared/src/components/MainFeedLayout.tsx (hunk 1 above)
Comment (rebelchris) [reply packages/shared/src/components/MainFeedLayout.tsx:295]: It also replaces `Most discussed feed` for users in the experiment basically.

File: packages/shared/src/components/sidebar/DiscoverSection.tsx (hunk 2 above)
Comment (rebelchris) [reply packages/shared/src/components/sidebar/DiscoverSection.tsx:85]: Correct yes, so the change is the one above.
If you're in the experiment you'll see `CommentFeed` (feed of comments) else you'll see `Feed` with most commented posts.

Comment (DragosIorgulescu) [reply packages/shared/src/components/sidebar/DiscoverSection.tsx:85]: nevermind, reading the PR description more carefully, I understand why this is here now 👍

File: packages/shared/src/hooks/useFeedLayout.ts (hunk 4 above)
Comment (rebelchris) [reply packages/shared/src/hooks/useFeedLayout.ts:32]: Hmm not sure, as it's also a sidebar condition?
Allocation would happen on mainLayout pretty much guaranteed anyway.

Comment (capJavert) [reply packages/shared/src/hooks/useFeedLayout.ts:32]: Yeah, since it is sidebar it is less relevant but I think cleaner vs adding dependencies to hook that can technically be used anywhere.

Comment (rebelchris) [reply packages/shared/src/hooks/useFeedLayout.ts:32]: Yeah agreed.
Think technically probably best to just have separate hook even.
Let's see how this experiment evolves

Comment (capJavert) [reply packages/shared/src/hooks/useFeedLayout.ts:32]: Solved it on slack. There were some other dependencies.

**Review by capJavert — APPROVED** [review]

File: packages/shared/src/components/sidebar/DiscoverSection.tsx [hunk 5]
```diff
@@ -15,6 +19,9 @@ export function DiscoverSection({
   enableSearch,
   ...defaultRenderSectionProps
 }: DiscoverSectionProps): ReactElement {
+  const hasCommentFeed = useFeature(feature.commentFeed);
+  const { checkHasCompleted, completeAction } = useActions();
+  const hasCompletedCommentFeed = checkHasCompleted(ActionType.CommentFeed);
```
Comment (capJavert) [inline packages/shared/src/components/sidebar/DiscoverSection.tsx:24]: @rebelchris you are missing `isActionsFetched, now the "New" label flashes on each page load since by default `checkHasCompleted` returns `false` until actions are fetched.
